react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+class ValidatedInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      password: '' // menu starts off as closed
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ password: event.target.value });
+  }
+
+  render() {
+    const password = this.state.password;
+    let icon = '';
+    let responseText = '';
+    if (!password) {
+      icon = 'fas fa-times';
+      responseText = 'A password is required, 8+ characters and at least ONE special character a special character (!, @, #, $, %, ^, &, *, (, or ))';
+    } else if (password.length < 8) {
+      icon = 'fas fa-times';
+      responseText = 'Password is too short, 8 characters or longer required';
+    } else if (password.length >= 8 && (!password.includes('@') || !password.includes('!') || !password.includes('#') || !password.includes('$') || !password.includes('%') || !password.includes('^') || !password.includes('&') || !password.includes('*') || !password.includes('(') || !password.includes(')'))) {
+      icon = 'fas fa-times';
+      responseText = 'Please add at least ONE special character (!, @, #, $, %, ^, &, *, (, or ))';
+    }
+    if (password.length >= 8 && (password.includes('@') || password.includes('!') || password.includes('#') || password.includes('$') || password.includes('%') || password.includes('^') || password.includes('&') || password.includes('*') || password.includes('(') || password.includes(')'))) {
```

Comment:
Try defining separate methods for these. When you have this much logic going on, it's much harder on your reader to understand what you are trying to do.

```js
hasSpecialCharacter() { // return true or false

}

isNotEmpty() {  // return true or false

}

isEightCharacters() { // return true or false

}
```